tests/templates/test_integration.py

```diff
@@ -488,6 +489,7 @@ def test_cv_template_followed_by_operations(self, template, diffable, nondiffabl
         diffable = {k: np.array(v) for k, v in diffable.items()}
 
         # Merge differentiable and non-differentiable arguments
+        nondiffable = nondiffable.copy()
```

Comment:
Out of curiosity...this was a bug besides tape mode, right? Was it just uncovered by switching on tape?

Reply:
> this was a bug besides tape mode, right? 

Yep!

> Was it just uncovered by switching on tape?

No, it was actually uncovered by me having pytest-randomly installed locally 😆 

It randomizes the order of the tests, so I was seeing failures due to this dictionary mutating randomly